pages/words/[id].js

```diff
@@ -22,24 +27,65 @@ export default function SingleEntry() {
             word: editedEntry.word,
             language: editedEntry.language,
             translated: editedEntry.translated,
+            notes: editedEntry.notes,
           };
         }
         return translation;
       })
     );
+    setIsShowMode(true);
   }
 
   return (
     <>
       <main>
-        <h1>Edit word</h1>
-        <CancelDeleteButton onClick={() => router.back()} />
-        {!entry ? null : (
-          <Form
-            entry={entry}
-            isEditMode={true}
-            onSubmitEvent={handleEditEntry}
-          />
+        {isShowMode ? (
+          <>
+            <h1>Word entry </h1>
+            <GoBackButton onClick={() => router.push("/words")} />
+            <EditButton onClick={() => setIsShowMode(false)}></EditButton>
+            {!entry ? null : (
+              <SingleEntry
+                isFavorite={entry.isFavorite}
+                word={entry.word}
+                language={entry.language}
+                translated={entry.translated}
+                notes={entry.notes}
+                onDeleteEntry={() => {
+                  router.push("/words");
+                  setTranslationList(
+                    translationList.filter(
+                      (translation) => translation.id !== id
+                    )
+                  );
+                }}
+                onToggleFavorite={() =>
+                  setTranslationList(
+                    translationList.map((translation) =>
+                      translation.id === id
+                        ? {
+                            ...translation,
+                            isFavorite: !translation.isFavorite,
+                          }
+                        : translation
+                    )
+                  )
+                }
```

Comment:
it would make sense to have function called `handleToggleFavorite` above the JSX where all of this logic is being handled. This will make your code more readable

Reply:
Thank you for your comments :-)
Here I tried already what you are suggesting. Finally I solved it like this, it was the only way to make it work. 
The problem is, that I have an object in the JSX (entry) not an array, so I always got the error "is not a function" when I tried to pass the handleToggle-function (I had the same problem with the delete-function.